python/beeai_framework/tools/search/wikipedia.py

```diff
@@ -13,16 +13,47 @@
 # limitations under the License.
 
 
+from typing import Any
+
+import wikipediaapi
+
+from pydantic import BaseModel, Field
+
+from beeai_framework.tools import ToolError
+from beeai_framework.tools.search import SearchToolOutput, SearchToolResult
 from beeai_framework.tools.tool import Tool
+from beeai_framework.utils import BeeLogger
+
+class WikipediaSearchToolInput(BaseModel):
+    query: str = Field(description="Search query, name of the Wikipedia page.")
+
+class WikipediaSearchToolResult(SearchToolResult):
+    pass
+
+class WikipediaSearchToolOutput(SearchToolOutput):
+    pass
 
 
-class WikipediaTool(Tool):
+class WikipediaSearchTool(Tool[WikipediaSearchToolInput]):
     name = "Wikipedia"
-    description = "Search factual and historical information, including biography, history, politics, geography, society, culture, science, technology, people, animal species, mathematics, and other subjects."  # noqa: E501
+    description = "Search factual and historical information, including biography, history, politics, geography, society, culture, science, technology, people, animal species, mathematics, and other subjects."
+    input_schema = WikipediaSearchToolInput
+    wiki_wiki = wikipediaapi.Wikipedia(user_agent="beeai-framework https://github.com/i-am-bee/beeai-framework", language='en')
```

Comment:
- What about just `client`?
- allow people to pass their options via the constructor (`__init__`)